x/pdv/handler.go

```diff
@@ -39,3 +42,23 @@ func handleMsgDistributeRewards(ctx sdk.Context, keeper Keeper, tokensKeeper tok
 
 	return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "Owner is not a Cerberus owner")
 }
+
+func handleMsgAccountRemove(ctx sdk.Context, keeper Keeper, tokensKeeper token.Keeper, msg MsgDeleteAccount) (*sdk.Result, error) {
+	for _, v := range keeper.GetSupervisors(ctx) {
+		addr, _ := sdk.AccAddressFromBech32(v)
+		if msg.Owner.Equals(addr) && !addr.Empty() {
+			tokensKeeper.SetBalance(ctx, msg.Owner, utils.InitialTokenBalance())
```

Comment:
this behaviour is `reset account` not `delete`

Reply:
Cosmos.SDK
```golang
// RemoveAccount removes an account for the account mapper store.
// NOTE: this will cause supply invariant violation if called
func (ak AccountKeeper) RemoveAccount(ctx sdk.Context, acc exported.Account) {
	addr := acc.GetAddress()
	store := ctx.KVStore(ak.key)
	store.Delete(types.AddressStoreKey(addr))
}
```
As far as I got the remove account operation is not safe. But you're right reset account sounds better.